cmasher/utils.py

```diff
@@ -233,6 +239,102 @@ def _get_cmap_perceptual_rank(
+) -> LinearSegmentedColormap:
+    """Create a composite matplotlib colormap by combining multiple colormaps.
+
+    Parameters:
+        cmaps (list[Union[Colormap, str]]): list of colormaps to combine.
+        nodes (List[float], optional): The blending points between
+            colormaps, in the range [0, 1]. Defaults to equal divisions.
+        n_rgb_levels (int, optional): Number of RGB levels for each colormap
+            segment. Defaults to 256.
+
+    Returns:
+        Colormap: The composite colormap.
+
+    Raises:
+        TypeError: If the list contains mixed datatypes or invalid
+            colormap names.
+        ValueError: If the cmaps contain only one single colormap,
+            or if the number of nodes is not one less than the number
+            of colormaps, or if the nodes do not contain incrementing values
+            between 0.0 and 1.0.
+
+    Note:
+        The colormaps are combined from bottom to top.
+
+    References:
+        - https://stackoverflow.com/questions/31051488/combining-two-matplotlib-colormaps/31052741#31052741
+
+    Examples:
+        # Example 1: Using colormap names
+        >>> cmaps = ["ocean", "prism", "coolwarm"]
+        >>> nodes = [0.2, 0.75]
+        >>> custom_cmap_1 = combine_cmaps(cmaps, nodes)
+
+        # Example 2: Using predefined colormaps
+        >>> cmap_0 = plt.get_cmap("Blues")
+        >>> cmap_1 = plt.get_cmap("Oranges")
+        >>> cmap_2 = plt.get_cmap("Greens")
+        >>> custom_cmap_2 = combine_cmaps([cmap_0, cmap_1, cmap_2])
+
+    """
+    # Check colormap datatype and convert to list[Colormap]
+    if len(cmaps) == 1:
+        raise ValueError("Cannot combine a single colormap.")
+    for cm in cmaps:
+        if not isinstance(cm, (Colormap, str)):
+            raise TypeError(f"Unsupported colormap type: {type(cm)}.")
+    _cmaps: list[Colormap] = [
+        cm if isinstance(cm, Colormap) else mpl.colormaps[cm] for cm in cmaps
+    ]
+
+    # Generate default nodes for equal separation
+    if nodes is None:
+        nodes = np.linspace(0, 1, len(_cmaps) + 1)[1:-1].tolist()
+
+    # Check nodes length
+    if len(nodes) != len(_cmaps) - 1:
+        raise ValueError(
+            "Number of nodes should be one less than the number of colormaps."
+        )
+
+    # Preprocess nodes list for easier segmentation
+    nodes.insert(0, 0.0)
+    nodes.append(1.0)
+
+    # Check node values
+    if not all(0.0 <= node <= 1.0 for node in nodes) or any(
+        nodes[i] >= nodes[i + 1] for i in range(len(nodes) - 1)
+    ):
+        raise ValueError(
+            "Nodes should contain incrementing values between 0.0 and 1.0."
+        )
+
+    # Generate composite colormap
+    combined_cmap_segments = []
+
+    for i, cmap in enumerate(_cmaps):
+        start_position = nodes[i]
+        end_position = nodes[i + 1]
+
+        # Calculate the length of the segment
+        segment_length = int(n_rgb_levels * (end_position - start_position))
```

Comment:
I'm not sure I understand why you're explicitly converting to `int` here. Isn't the result already an int ?

Reply:
I think `n_rgb_levels * (end_position - start_position)` may or may not be an interger here, because `n_rgb_levels` and `end_position - start_position` could be any combination?
`n_rgb_levels` could be any positive interger and `end_position - start_position` could be any float within range (0, 1).

Reply:
Oh you're right, of course ! Sorry I think I got confused while reading it, my bad !

Reply:
Quote the wisdom man (@neutrinoceros):
> no need to apologise !